### PR TITLE
jssrc2cpg: fix call receiver and base handling

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
This is for: https://github.com/joernio/joern/issues/1685

### CODE REVIEW

File: joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 1]
```diff
@@ -515,6 +513,44 @@ class SimpleAstCreationPassTest extends AbstractPassTest {
...
+
+      val List(receiver) = call.receiver.isCall.l
+      receiver.code shouldBe "a.b"
+      receiver.methodFullName shouldBe Operators.fieldAccess
+
+      val List(receiverAsArg) = call.argument(0).cast[Call].l
+      receiver shouldBe receiverAsArg
+
+      val List(base) = receiver.astChildren.isIdentifier.argumentIndex(1).l
+      base.name shouldBe "a"
```
Comment (maltek) [inline joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:537]: I don't think that's we want yet.

We want

- either the `a` identifier as arg 0 - apparently that's how it's always been with js2cpg
- or the `a.b` field access as arg 0 - that matches the original code more precisely, but I don't know if there was a reason why we've gone with the identifier previously
- but neither can be arg 1. That must be reserved for the "x" identifier that's actually in position 1 in the original code.

**Review by maltek — CHANGES_REQUESTED** [review]

File: joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:537]: Look closely. Thats the base of the receiver (a field access call here).

Comment (maltek) [reply joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:537]: oh, I see :facepalm:

File: joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 2]
```diff
@@ -515,6 +513,44 @@ class SimpleAstCreationPassTest extends AbstractPassTest {
...
+      val List(accessElement) = receiver.astChildren.isFieldIdentifier.argumentIndex(2).l
+      accessElement.canonicalName shouldBe "b"
+
+      val List(recvArg, baseArg, arg) = call.argument.l.sortBy(_.order)
+      recvArg shouldBe receiver
+      recvArg.argumentIndex shouldBe 0
+      recvArg.order shouldBe 0
+      baseArg.code shouldBe "a"
+      baseArg.argumentIndex shouldBe 0
+      baseArg.order shouldBe 1
```
Comment (maltek) [inline joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:548]: When tracking flows into parameter 0 of the callee of such a call, I think we'd get duplicate dataflows here - one through `recvArg` and another through `baseArg`.

My vote would be to just get rid of the `baseArg` - but I don't know the reason why we generated the `baseArg` in js2cpg to begin with.

**Review by maltek — APPROVED** [review]

**Pull request merged**